### PR TITLE
Repair: make repair system keyspace first more robust

### LINKED ISSUE #3543: Repair: don't use whitelist as the only source of internal tables
Right now, when SM creates table repair order, it distinguishes internal keyspaces from regular ones only via predefined list (which also gives different priorities to different internal tables). If keyspace cannot be found in this list, SM treats it as a regular one. In this case, it would be better if SM also checked for `system` prefix. Right now this list might be enough, but it would be nice to make the ordering process more robust in case a new internal keyspace is introduced in Scylla.

### PR DESCRIPTION
**Pull request by Michal-Leszczynski**
Fixes #3543

### CODE REVIEW

File: pkg/service/repair/preference.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+func NewInternalTablePreference() TablePreference {
+	return internalTablePreference{
+		{
+			Keyspace: "system_auth",
+			Tables: []string{
+				"role_attributes",
+				"role_members",
+			},
+		},
+		{Keyspace: "system_traces"},
```
Comment (mykaul) [inline pkg/service/repair/preference.go:28]: nit: system_traces can be the last, it's less important than any other table.

File: pkg/service/repair/preference.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+			if t == t1 {
+				return true
+			}
+			if t == t2 {
+				return false
+			}
+		}
+	}
+	return false
+}
```
Comment (karol-kokoszka) [inline pkg/service/repair/preference.go:74]: Could be easily unit tested.
Please add test cases for KLess and TLess.

File: pkg/service/repair/preference.go (hunk 1 above)
Comment (Michal-Leszczynski) [reply pkg/service/repair/preference.go]: done

File: pkg/service/repair/preference.go (hunk 2 above)
Comment (Michal-Leszczynski) [reply pkg/service/repair/preference.go:74]: done

**Pull request merged**